### DISCUSSION TITLE
Hook items.read as filter?

### DISCUSSION THREAD

**Discussion by gnimmelf** [directus/directus#7837] [Feature Requests]
Hi guys,

Reading the documentation on [api-hooks](https://docs.directus.io/guides/api-hooks/)
and looking at issues
#3736
#4847

When the query is passed instead of the items, shouldn't returning an altered query be reflected in the result-set? - The create-hooks allows you to modify the payload, so wouldn't it be natural for the `items.read`-hook to also let you modify the query?

The issue is same as #4847, which requires a filter beyond what can be currently set on roles & permissions. - Incidentally #4847 doesn't seem like a duplicate of #3736 as that one just ask for a hook.

A bit on the side, but this also ties into #7744, which in essence is asking for a better way to understand the hooks. For example, I am guessing now that since payload comes in as the first argument for `*.create:before`-hooks, then it is also the return value? 

Use case:
I have `directus_users` m2o `vendor` o2m `products` and I need to filter image files for the product gallery to only the current vendor.

Cheers!

**Comment by rijkvanzanten** [comment]
Makes sense to me 👍🏻

**Reply by gnimmelf** [reply]
Nice!

I can create an issue from this if you like? - Bug or a feature request?

**Reply by rijkvanzanten** [reply]
Lets hold off a quick sec while we figure out https://github.com/directus/directus/pull/7853, https://github.com/directus/directus/discussions/7744, https://github.com/directus/directus/discussions/7843, and https://github.com/directus/directus/pull/7906#issuecomment-915304854 🤔

**Reply by rijkvanzanten** [reply]
(All links documenting the same underlying situation, not 4 different things that block this)

**Comment by sapientorius** [comment]
In the documentation this makes also no sense: action('items.read', adminOnly);

Actions fire after the event, so you cannot check for permissions if there is no "read before" event.

Is there any workaround where you can check permissions or modify the return object before sending it back to the client?

**Comment by benhaynes** [comment]
This feature has been approved, and will be tracked here: https://github.com/directus/directus/issues/10164

**Comment by nickrum** [comment]
This has been added in #10279.
